include/dlaf/matrix/distribution.h

```diff
@@ -490,6 +501,27 @@ class Distribution {
             localElementDistanceFromLocalTile<Coord::Col>(begin.col(), end.col())};
   }
 
+  /// TODO
+  GlobalTileIndex globalTileIndexFromSubDistribution(const GlobalElementIndex& sub_offset,
+                                                     const Distribution& sub_distribution,
+                                                     const GlobalTileIndex& sub_index) const noexcept {
+    DLAF_ASSERT(sub_index.isIn(sub_distribution.nrTiles()), sub_index, sub_distribution.nrTiles());
+    DLAF_ASSERT(isCompatibleSubDistribution(sub_offset, sub_distribution), "");
+    const GlobalTileIndex tile_offset = globalTileIndex(sub_offset);
+    return tile_offset + common::sizeFromOrigin(sub_index);
+  }
+
+  TileElementIndex tileElementOffsetFromSubDistribution(
+      const GlobalElementIndex& sub_offset, const Distribution& sub_distribution,
+      const GlobalTileIndex& sub_index) const noexcept {
+    DLAF_ASSERT(sub_index.isIn(sub_distribution.nrTiles()), sub_index, sub_distribution.nrTiles());
+    DLAF_ASSERT(isCompatibleSubDistribution(sub_offset, sub_distribution), "");
+    return {
+        sub_index.row() == 0 ? tileElementFromGlobalElement<Coord::Row>(sub_offset.row()) : 0,
+        sub_index.col() == 0 ? tileElementFromGlobalElement<Coord::Col>(sub_offset.col()) : 0,
+    };
+  }
```

Comment:
I've currently included a minimal set of helper functions as methods on `Distribution`. This is sufficient, together with `MatrixRef` storing the original `Matrix` (which gives access to its `Distribution`), `MatrixRef` inheriting `MatrixBase` (which gives access to the sub-distribution), and `MatrixRef` storing the offset between the original distribution and the sub-distribution, thus keeping track of the relationship between the two. This is why the helpers here take the sub-distribution and the offset. 

An alternative would be to have some kind of helper which keeps track of this relationship and performs necessary index transformations/calculations. Do you think this would be more readable and/or maintainable?

Reply:
Let's see in real usage if this needs refactoring, but I'm keeping this as is for now.